src/main/kotlin/org/jetbrains/research/ml/coding/assistant/dataset/TaskTrackerDatasetFetcher.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.ml.coding.assistant.dataset
+
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import org.jetbrains.research.ml.coding.assistant.dataset.model.*
+import org.jetbrains.research.ml.coding.assistant.utils.getListFiles
+import java.io.File
+import kotlin.streams.toList
+
+object TaskTrackerDatasetFetcher : DatasetFetcher {
+    override fun fetchDataset(file: File): Dataset {
+        require(file.isDirectory) { "Argument has to be directory with tasks" }
+        val taskSolutions = file.getListFiles().parallelStream()
+            .filter { it.isDirectory }
+            .map(this::fetchTaskSolutions).toList()
+        return Dataset(taskSolutions)
+    }
+
+    override fun fetchTaskSolutions(file: File): TaskSolutions {
+        require(file.isDirectory) { "Argument has to be directory with solution files" }
+        val solutions = file.getListFiles().parallelStream()
+            .map(this::fetchDynamicSolution)
+            .filter { it.hasFinalSolution() }
+        return TaskSolutions(
+            datasetTask = DatasetTask.createFromString(file.name),
+            solutions?.toList() ?: listOf()
+        )
+    }
+
+    private fun fetchDynamicSolution(file: File): DynamicSolution {
+        require(file.isFile && file.extension == "csv") { "The file has to be csv" }
```

Comment:
We already have the `Extension` enum class in the  `FileTestUtil` file. I think it would be great if you move it into src utils and add CSV exstention